### PR TITLE
STONEBLD-783: set on-target-branch to default branch properly

### PR DESCRIPTION
**Pull request by tkdchen**

### CODE REVIEW

File: controllers/component_build_controller_pac.go [hunk 1]
```diff
@@ -681,12 +703,22 @@ func (r *ComponentBuildReconciler) UnconfigureRepositoryForPaC(log logr.Logger,
...
-func generatePaCPipelineRunForComponent(component *appstudiov1alpha1.Component, pipelineSpec *tektonapi.PipelineSpec, additionalPipelineParams []tektonapi.Param, onPull bool) (*tektonapi.PipelineRun, error) {
+func generatePaCPipelineRunForComponent(
+	component *appstudiov1alpha1.Component,
+	pipelineSpec *tektonapi.PipelineSpec,
+	additionalPipelineParams []tektonapi.Param,
+	onPull bool,
+	targetBranch string) (*tektonapi.PipelineRun, error) {
+
 	var targetBranches []string
 	if component.Spec.Source.GitSource != nil && component.Spec.Source.GitSource.Revision != "" {
```
Comment (Michkov) [inline controllers/component_build_controller_pac.go:714]: move this to selection of target branch -> it's strange that we pass targetBranch to the function but it's not being used.

Comment (tkdchen) [reply controllers/component_build_controller_pac.go]: Done

File: controllers/component_build_controller_pac.go [hunk 2]
```diff
@@ -449,8 +458,11 @@ func (r *ComponentBuildReconciler) ConfigureRepositoryForPaC(ctx context.Context
 	authorName := "redhat-appstudio"
 	authorEmail := "appstudio@redhat.com"
 
+	var pacTargetBranch string
+
 	if component.Spec.Source.GitSource != nil && component.Spec.Source.GitSource.Revision != "" {
 		baseBranch = component.Spec.Source.GitSource.Revision
+		pacTargetBranch = component.Spec.Source.GitSource.Revision
```
Comment (Michkov) [inline controllers/component_build_controller_pac.go:465]: can we merge this with baseBranch? That would safe us a github/gitlab call.

File: controllers/component_build_controller_pac.go [hunk 3]
```diff
@@ -394,42 +394,51 @@ func (r *ComponentBuildReconciler) ensurePaCRepository(ctx context.Context, comp
 	return nil
 }
 
-// ConfigureRepositoryForPaC creates a merge request with initial Pipelines as Code configuration
-// and configures a webhook to notify in-cluster PaC unless application (on the repository side) is used.
-func (r *ComponentBuildReconciler) ConfigureRepositoryForPaC(ctx context.Context, component *appstudiov1alpha1.Component, config map[string][]byte, webhookTargetUrl, webhookSecret string) (prUrl string, err error) {
-	log := r.Log.WithValues("Repository", component.Spec.Source.GitSource.URL)
+func (r *ComponentBuildReconciler) generatePaCPipelineRunConfigs(
```
[the remark this answers is not in the document]
Comment (tkdchen) [reply controllers/component_build_controller_pac.go:399]: Added.

File: controllers/component_build_controller_pac.go (hunk 2 above)
Comment (tkdchen) [reply controllers/component_build_controller_pac.go:467]: I'm not sure I get your point. Can you be more specific?

Comment (Michkov) [reply controllers/component_build_controller_pac.go:467]: baseBranch and pacTargetBranch is the same value, when base branch is "" then it's detected, which is exactly what we are doing. So you should be able to use baseBranch and just getDefault branch in this part of code.

Comment (tkdchen) [reply controllers/component_build_controller_pac.go]: Done. PTAL.

**Review by Michkov — APPROVED** [review]

**Review by mmorhun — APPROVED** [review]

**Review by psturc — APPROVED** [review]
verified with [e2e-test](https://github.com/redhat-appstudio/e2e-tests/pull/367) that I updated by changing the default branch (to something else then `master`/`main`)

File: controllers/component_build_controller_pac.go [hunk 4]
```diff
@@ -484,6 +495,17 @@ func (r *ComponentBuildReconciler) ConfigureRepositoryForPaC(ctx context.Context
...
 		}
 
+		if baseBranch == "" {
+			baseBranch, err = github.GetDefaultBranch(ghclient, owner, repository)
+			if err != nil {
+				return "", nil
+			}
+		}
+
+		prOnPushYaml, prOnPRYaml, err := r.generatePaCPipelineRunConfigs(ctx, log, component, baseBranch)
```
[the remark this answers is not in the document]
Comment (tkdchen) [reply controllers/component_build_controller_pac.go]: Done

File: pkg/github/github_helper.go [hunk 5]
```diff
@@ -30,6 +30,9 @@ var CreatePaCPullRequest func(g *GithubClient, d *PaCPullRequestData) (string, e
 var UndoPaCPullRequest func(g *GithubClient, d *PaCPullRequestData) (string, error) = undoPaCPullRequest
 var SetupPaCWebhook func(g *GithubClient, webhookUrl, webhookSecret, owner, repository string) error = setupPaCWebhook
 var DeletePaCWebhook func(g *GithubClient, webhookUrl, owner, repository string) error = deletePaCWebhook
+var GetDefaultBranch func(*GithubClient, string, string) (string, error) = func(client *GithubClient, owner string, repository string) (string, error) {
```
[the remark this answers is not in the document]
Comment (tkdchen) [reply pkg/github/github_helper.go]: Done

**Pull request merged**